shell/platform/embedder/tests/embedder_unittests_gl.cc

```diff
@@ -2028,6 +2028,8 @@ TEST_F(EmbedderTest,
   frame_latch.Wait();
 
   ASSERT_EQ(frames_expected, frames_seen);
+
+  FlutterEngineShutdown(engine.release());
```

Comment:
Is a similar fix needed anywhere aside from these two?

Reply:
This is sort of a catchall fix and might work if there are similar problems in the other frame tests, but it's hard to tell without the trace.
It might be a good idea for us to wait on merging this PR before https://github.com/flutter/engine/pull/28274 is merged, so that we can verify if there's an internal engine state problem rather than a reference to the test stack getting accessed like I suspect. And if it's the latter problem, there might be a more specific/better fix.

Did 1100 test runs off CI with the NVIDIA driver (GM107GL [Quadro K1200]) and couldn't reproduce it.

Reply:
Will merge this in for now as it's safe. Not certain that it fixes the flake but I'll fix it if it returns.